crates/biome_js_analyze/src/lint/nursery/missing_playwright_await.rs

```diff
@@ -0,0 +1,501 @@
+    "toHaveAttribute",
+    "toHaveClass",
+    "toHaveCount",
+    "toHaveCSS",
+    "toHaveId",
+    "toHaveJSProperty",
+    "toHaveScreenshot",
+    "toHaveText",
+    "toHaveTitle",
+    "toHaveURL",
+    "toHaveValue",
+    "toHaveValues",
+    "toContainClass",
+    "toPass",
+];
+
+pub enum MissingAwaitType {
+    ExpectMatcher(TokenText),
+    ExpectPoll,
+    TestStep,
+}
+
+impl Rule for MissingPlaywrightAwait {
+    type Query = Ast<JsCallExpression>;
+    type State = MissingAwaitType;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let call_expr = ctx.query();
+
+        // Check for test.step() calls
+        if is_test_step_call(call_expr) {
+            if !is_properly_handled(call_expr) {
+                return Some(MissingAwaitType::TestStep);
+            }
+            return None;
+        }
+
+        // Check for expect calls with async matchers
+        if let Some(matcher_name) = get_async_expect_matcher(call_expr)
+            && !is_properly_handled(call_expr)
+        {
+            return Some(matcher_name);
+        }
+
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        match state {
+            MissingAwaitType::ExpectMatcher(matcher) => {
+                let matcher_text = matcher.text();
+                Some(
+                    RuleDiagnostic::new(
+                        rule_category!(),
+                        node.range(),
+                        markup! {
+                            "Async matcher "<Emphasis>{matcher_text}</Emphasis>" must be awaited or returned."
+                        },
+                    )
+                    .note(markup! {
+                        "Add "<Emphasis>"await"</Emphasis>" before the expect call or return the promise."
+                    }),
+                )
+            }
+            MissingAwaitType::ExpectPoll => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    node.range(),
+                    markup! {
+                        <Emphasis>"expect.poll"</Emphasis>" must be awaited or returned."
+                    },
+                )
+                .note(markup! {
+                    "Add "<Emphasis>"await"</Emphasis>" before the expect call or return the promise."
+                }),
+            ),
+            MissingAwaitType::TestStep => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    node.range(),
+                    markup! {
+                        <Emphasis>"test.step"</Emphasis>" must be awaited or returned."
+                    },
+                )
+                .note(markup! {
+                    "Add "<Emphasis>"await"</Emphasis>" before the test.step call or return the promise."
+                }),
```

Comment:
When a rule has an action, we should put the "how to fix it" part in the action's message instead.